eng/tools/generator/cmd/v2/common/cmdProcessor.go

```diff
@@ -210,29 +213,56 @@ func ExecuteTspClient(path string, args ...string) error {
 		return err
 	}
 
-	var buf bytes.Buffer
-	if _, err = io.Copy(&buf, stderr); err != nil {
+	var stdoutBuffer bytes.Buffer
+	if _, err = io.Copy(&stdoutBuffer, stdoutPipe); err != nil {
 		return err
 	}
 
-	if err := cmd.Wait(); err != nil || buf.Len() > 0 {
-		if buf.Len() > 0 {
-			log.Println(buf.String())
+	var stderrBuffer bytes.Buffer
+	if _, err = io.Copy(&stderrBuffer, stderrPipe); err != nil {
+		return err
+	}
+
+	cmdWaitErr := cmd.Wait()
+	fmt.Println(stdoutBuffer.String())
+
+	if cmdWaitErr != nil || stderrBuffer.Len() > 0 {
+		if stderrBuffer.Len() > 0 {
+			log.Println(stderrBuffer.String())
 
 			// filter npm notice log
-			lines := strings.Split(buf.String(), "\n")
-			newErrInfo := make([]string, 0, len(lines))
-			for _, line := range lines {
+			newErrMsgs := make([]string, 0)
+			for _, line := range strings.Split(stderrBuffer.String(), "\n") {
 				if len(strings.TrimSpace(line)) == 0 {
 					continue
 				}
 				if !strings.Contains(line, "npm notice") {
-					newErrInfo = append(newErrInfo, line)
+					newErrMsgs = append(newErrMsgs, line)
+				}
+			}
+
+			// filter diagnostic errors
+			if len(newErrMsgs) == 1 &&
```

Comment:
Should we check `if len(newErrMsgs) > 1 && ..`?

Reply:
>= 1